app/presenters/document_presenter.rb

```diff
@@ -8,13 +8,31 @@ class DocumentPresenter
     :bulk_published,
     to: :"document.details"
 
+  attr_reader :finder
+
   def initialize(finder, document)
     @finder = finder
     @document = document
   end
 
+  def respond_to?(method)
+   if finder.facets.map(&:key).include?(method.to_s)
+     true
+   else
+     super
+   end
+ end
+
+  def method_missing(method, *args)
+    if finder.facets.map(&:key).include?(method.to_s)
+      document.details.try(method)
```

Comment:
Actually, my previous comment makes no sense – `try` is to guard against `details` being `nil` here. Will that ever be the case?